### ISSUE TITLE
Concordance index errors with repeated events

### ISSUE THREAD

**Issue by hlin117** [CamDavidsonPilon/lifelines#236]
The following produces a `nan` value in lifelines version `0.9.1.0`.

```
>>> import lifelines
>>> import numpy as np
>>> from lifelines.utils import concordance_index
>>> times = np.arange(3)
>>> events = np.ones(3)
>>> concordance_index(events, times)
nan
```

I don't think that's the expected outcome, it should be 0.5, right?

The same error happens when `n=1`. In this case, I think a `ValueError` should be thrown for input validation though.

```
>>> import lifelines
>>> import numpy as np
>>> from lifelines.utils import concordance_index
>>> times = np.arange(1)
>>> events = np.ones(1)
>>> concordance_index(events, times)
nan
```

**Comment by hlin117** [comment]
I also think that the following below should return `2.5 / 3 = 0.8333` instead of 1.0

```
>>> from lifelines.utils import concordance_index
>>> events = [1, 1, 2]
>>> times = [1, 2, 3]
>>> concordance_index(events, times)
1.0
```

**Comment by CamDavidsonPilon** [comment]
Hey there, I'm investigating this, and ty for bring this to my attention. I can offer some advice atm. Your script:

```
>>> import lifelines
>>> import numpy as np
>>> from lifelines.utils import concordance_index
>>> times = np.arange(3)
>>> events = np.ones(3)
>>> concordance_index(events, times)
nan
```

is a bit of a pathological case for the concordance index. In this case, all actual events happen at the same time and all events  have been observed. So the concordance index will say "hey, when eventA happens _before_ eventB, does that hold true for our predicted data?" Now, if eventA == eventB for all events, then the initial condition (eventA happens _before_ eventB) is never able to be evaluated. What we get is actually no admissible events to compare (technically speaking, we get a division by 0 error in https://github.com/CamDavidsonPilon/lifelines/blob/98e3923d181891eb147a8c8aaa03fa07f00c0ec6/lifelines/utils/__init__.py#L916). 

Returning `nan` is wrong though, I think a ValueError or something should be thrown instead. 

Another pathological case is when `n==1` - in this case, there are no possible pairs, so a division by zero also occurs. 

-------
For your second comment, can you explain why you think 2.5/3 should be the right answer?

**Comment by hlin117** [comment]
Hey Cameron, thanks for the reply. 

> For your second comment, can you explain why you think 2.5/3 should be the right answer?

I was using the definition from this [nature paper](http://www.nature.com/nbt/journal/v32/n12/full/nbt.2877.html), shown here:
<img width="660" alt="screenshot 2016-05-23 10 53 59" src="https://cloud.githubusercontent.com/assets/1865885/15476603/8ceb9690-20d5-11e6-98fe-56e5a053b80e.png">

Using this definition, if `events = [1, 1, 2]` and `times = [1, 2, 3]`, it makes more sense that the concordance index = `2.5 / 3`. 

The definition from the nature paper doesn't match the paper from [this paper](http://papers.nips.cc/paper/3375-on-ranking-in-survival-analysis-bounds-on-the-concordance-index.pdf) though. 

<img width="677" alt="screenshot 2016-05-23 11 07 54" src="https://cloud.githubusercontent.com/assets/1865885/15476765/a37a0fd0-20d6-11e6-9974-0e6f9f5b10c1.png">

I'm willing to believe that the nature paper tried to imply that the `0.5` would be double counted to produce `1.0`, but it did so using incorrect mathematical notation. So leaving `ci([1, 1, 2], [1, 2, 3]) = 1.0` seems fine to me.

**Comment by CamDavidsonPilon** [comment]
So I've added an ZeroDivisionError to the latest version of pypi, 0.9.2. I've left the c-index as is, according to the second paper there.

**Comment by hlin117** [comment]
Great, thanks @CamDavidsonPilon!

**Comment by AyuranS** [comment]
Hi @CamDavidsonPilon I think I have spotted an error in lifelines where I am trying to calculate the concordance_index of the results of a different predictive model that predicts the same survival times for multiple users. None of the events are censored but the code just continues to run in a loop without providing an error or timing out.

**Comment by CamDavidsonPilon** [comment]
Thanks for the report, @AyuranS - are you able to create a reproducible example?

**Comment by AyuranS** [comment]
@CamDavidsonPilon , I realised on trying to reproduce the bug, the issue was due to having some NaNs in the actual Time To Event column when the input is a DataFrame. This doesn't really make it a use case for the function anymore but maybe the input error isn't being handled correctly.
I believe this would reproduce the error at least on my machine.
`test1= pd.DataFrame(np.random.randint(800, size=800), columns=['prediction'])
test2= pd.DataFrame(np.random.randint(800, size=800), columns=['actuals'])
test2[-50:-1]=np.nan
print(test1.shape)
print(test2.shape)
print(lf.utils.concordance_index(test2, test1))`
